File: cuiserRice.cs
using System;

class CuiseurARiz
{
    private bool cuissonEnCours = false;
    private bool maintienAuChaud = false;
    private int tempsMaintienAuChaud = 30;
    private int surchauffeTemperature = 120;

    public void CuireRiz()
    {
        Console.WriteLine("Cuisson en cours...");
        cuissonEnCours = true;
        try
        {
            for (int i = 0; i < 15; i++)
            {
                System.Threading.Thread.Sleep(1000);
                if (SurchauffeDetectee())
                    throw new Exception("Surchauffe détectée. Arrêt de la cuisson.");
            }
            if (!QualiteDuRiz())
                throw new Exception("Le riz ne répond pas aux critères de qualité.");
        }
        catch (Exception e)
        {
            Console.WriteLine($"Erreur pendant la cuisson : {e.Message}");
        }
        finally
        {
            cuissonEnCours = false;
        }
    }

    public void MaintienChaud()
    {
        if (!cuissonEnCours)
        {
            throw new Exception("Aucune cuisson en cours. Commencez d'abord la cuisson.");
        }

        Console.WriteLine($"Le riz est maintenant maintenu au chaud pendant {tempsMaintienAuChaud} secondes.");
        maintienAuChaud = true;
        System.Threading.Thread.Sleep(tempsMaintienAuChaud * 1000);
        Console.WriteLine("Le maintien au chaud est terminé.");
        maintienAuChaud = false;
    }

    private bool SurchauffeDetectee()
    {
        int temperatureActuelle = 130;
        if (temperatureActuelle >= surchauffeTemperature)
        {
            throw new Exception("Surchauffe détectée. Arrêt de la cuisson.");
        }
        return false;
    }

    private bool QualiteDuRiz()
    {
        return true;
    }
}

class Program
{
    static void Main()
    {
        CuiseurARiz cuiseur = new CuiseurARiz();

        try
        {
            cuiseur.CuireRiz();
        }
        catch (Exception e)
        {
            Console.WriteLine($"Erreur générale pendant la cuisson : {e.Message}");
        }

        try
        {
            cuiseur.MaintienChaud();
        }
        catch (Exception e)
        {
            Console.WriteLine($"Erreur générale pendant le maintien au chaud : {e.Message}");
        }
    }
}

File: cuiserRice.py
import time

class CuiseurARiz:
    def __init__(self):
        self.cuisson_en_cours = False
        self.maintien_au_chaud = False
        self.temps_maintien_au_chaud = 30
        self.surchauffe_temperature = 120

    def cuire_riz(self):
        try:
            print("Cuisson en cours...")
            self.cuisson_en_cours = True

            for _ in range(15):
                time.sleep(1)
                if self.surchauffe_detectee():
                    raise Exception("Surchauffe détectée. Arrêt de la cuisson.")
            if not self.qualite_du_riz():
                raise Exception("Le riz ne répond pas aux critères de qualité.")
        except Exception as e:
            print(f"Erreur pendant la cuisson : {str(e)}")
        finally:
            self.cuisson_en_cours = False

    def maintien_chaud(self):
        try:
            if self.cuisson_en_cours:
                print(f"Le riz est maintenant maintenu au chaud pendant {self.temps_maintien_au_chaud} secondes.")
                self.maintien_au_chaud = True
                time.sleep(self.temps_maintien_au_chaud)
                print("Le maintien au chaud est terminé.")
                self.maintien_au_chaud = False
            else:
                raise Exception("Aucune cuisson en cours. Commencez d'abord la cuisson.")
        except Exception as e:
            print(f"Erreur pendant le maintien au chaud : {str(e)}")

    def surchauffe_detectee(self):
        temperature_actuelle = 130
        if temperature_actuelle >= self.surchauffe_temperature:
            return True
        return False

    def qualite_du_riz(self):
        return True
